src/main/java/uk/gov/justice/hmpps/prison/service/AppointmentsService.kt

```diff
@@ -171,37 +174,54 @@ class AppointmentsService(
   @Transactional
   fun deleteBookingAppointments(appointmentIds: List<Long>) {
     appointmentIds.forEach { appointmentId: Long ->
-      offenderIndividualScheduleRepository.findByIdOrNull(appointmentId)?.let { appointment ->
-        offenderIndividualScheduleRepository.deleteById(appointmentId)
-        trackAppointmentDeletion(appointment)
+      try {
+        offenderIndividualScheduleRepository.findWithLockById(appointmentId).ifPresent { appointment ->
+          offenderIndividualScheduleRepository.deleteById(appointmentId)
+          trackAppointmentDeletion(appointment)
+        }
+      } catch (e: CannotAcquireLockException) {
+        throw processLockError(e, appointmentId)
```

Comment:
Here it fails if any of the appointments are locked - think thats ok as its a rare event